### PR TITLE
add HJ-minlo ptcut card

### PR DESCRIPTION
**Pull request by nsmith-**

### CODE REVIEW

**Comment by agrohsje** [comment]
Hi Nick,
I suggest to put the card here , where the other hj minlo files are stored: 
https://github.com/cms-sw/genproductions/tree/master/bin/Powheg/production/2017/13TeV/Higgs

File: bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ncall1   150000    ! number of calls for initializing the integration grid
+itmx1    1        ! number of iterations for initializing the integration grid
+ncall2   100000   ! number of calls for computing the integral and finding upper bound
+itmx2    5        ! number of iterations for computing the integral and finding upper bound
+fastbtlbound 1    ! (default 0) if 1 use fast btilde bound
+storemintupb 1    ! store calls for grids
+foldcsi   2       ! number of folds on csi integration
+foldy     2       ! number of folds on  y  integration
+foldphi   2       ! number of folds on phi integration
+nubound 2000    ! number of bbarra calls to setup norm of upper bounding function
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:34]: did you test that the settings are high enough to get good results? i see that we used them in the other card i mentioned. however, that doesn't mean much, so i prefer to ask.

File: bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+! OPTIONAL PARAMETERS
+
+testplots  1       ! (default 0, do not) do NLO and PWHG distributions
+
+withnegweights 1  ! (1 default) If 1 output negative weighted events.
+                   ! If 0 descard them
+
+renscfact  1d0   ! (default 1d0) ren scale factor: muren  = muref * renscfact 
+facscfact  1d0   ! (default 1d0) fac scale factor: mufact = muref * facscfact 
+pdfreweight 0       ! PDF reweighting
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:45]: centrally handled. suggest to remove.

File: bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+bornktmin 200.0    ! Minimum transverse momentum of the Higgs at the underlying Born level
+
+storeinfo_rwgt 0   ! store info to allow for reweighting
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:49]: same here. suggest to remove.

File: bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+iseed    SEED       ! initialize random number sequence 
+
+# variables for including quark mass effects
+quarkmasseffects 1   ! Include quark mass effects
+topmass 172.5d0      ! top quark mass (172.5 in HNNLO-patches/mdata.f)
+bottommass 3.38d0    ! bottom quark mass in MSbar at MH
+bmass_in_minlo 1     ! Include quark mass effects in Sudakov exponent
+
+fakevirt 1
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:67]: did you check the impact from using fakevirt?

File: bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+renscfact  1d0   ! (default 1d0) ren scale factor: muren  = muref * renscfact 
+facscfact  1d0   ! (default 1d0) fac scale factor: mufact = muref * facscfact 
+pdfreweight 0       ! PDF reweighting
+
+bornktmin 200.0    ! Minimum transverse momentum of the Higgs at the underlying Born level
+
+storeinfo_rwgt 0   ! store info to allow for reweighting
+
+flg_debug      1   ! store extra event info for debugging
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:51]: do you need that info ?

File: bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input (hunk 1 above)
Comment (nsmith-) [reply bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:34]: I started from https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/production/2017/13TeV/Higgs/HJ_MiNLO_NNLOPS_NNPDF31_13TeV/HJ_MiNLO_NNLOPS_NNPDF31_13TeV.input and simply adjusted the `bornktmin`. Then I ran a small test of 5k events with good results for the main quantity of interest
image

Comment (agrohsje) [reply bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:34]: Thanks @nsmith- !

File: bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input (hunk 4 above)
Comment (nsmith-) [reply bin/Powheg/production/2017/13TeV/HJ_MINLO_Pt-200ToInf/HJ_MiNLO_NNPDF31_13TeV.input:67]: Somewhere in the printout I noticed that powheg says `powheginput WARNING: unused variable fakevirt` so I guess it is superfluous (was in old card) Will remove and re-test

**Comment by agrohsje** [comment]
Thanks @nsmith- . Yes, please do so. Also in single top Powheg we have settings that don't belong there. This is totally annoying but during review not always possible to catch as all processes are different and I don't know all settings for all processes by heart.

**Comment by nsmith-** [comment]
Ok so actually somehow these changes ended up creating the error on running the gridpack:
```
rwgt_info_addweight:  rwgt_info_addweight: too many weights, increase maxweights
```
I'm not sure what is going on, as I didn't change anything with the powheg compilation part since earlier

**Comment by agrohsje** [comment]
hi nick. the error is not related to our changes. that comes from the pdf weights you are evaluating. i think a quick fix would be to clean the pwg-rwl.dat file up. this is common choice for cms. i am sure you can remove some of them and you will never suffer. you can even do this after creating the gridpack. alternativ is to change the code : 
pwhg_rwl.h:      integer, parameter :: rwl_maxweights=1200,rwl_maxgroups=50
but i would really check the file. i doubt you need all the pdfs listed there. let me know.

**Comment by nsmith-** [comment]
Ah yeah I had edited it down for a test and that worked fine, but I thought the default was the "official" pdf requirements. If I'm free to change to whatever I want then I can pare it down quite a bit

**Comment by agrohsje** [comment]
This was the official agreement with the PDF forum but there is a lot of redundancy and backups. Feel free to clean-up a bit.

**Pull request merged**